assets/src/js/listing_pages.js

```diff
@@ -7,29 +7,64 @@ export const setupListingPages = () => {
     return;
   }
 
-  // Setup behaviour for list/grid toggle.
-  const listViewToggle = document.querySelector('.list-view-toggle');
-  const gridViewToggle = document.querySelector('.grid-view-toggle');
+  const toggleButton = document.querySelector('.layout-toggle');
 
-  if (!listViewToggle && !gridViewToggle) {
-    return;
-  }
+  const clearStorageAfter = 30 * 60 * 1000; // 30mins in milliseconds
+
+  // Function to clear localStorage after a set time
+  const clearLocalStorage = () => {
+    setTimeout(() => {
+      localStorage.removeItem('layout');
+    }, clearStorageAfter);
+  };
 
-  listingPageContent.classList.toggle('wp-block-query--list', gridViewToggle);
-  listingPageContent.classList.toggle('wp-block-query--grid', listViewToggle);
+  const switchViews = layoutView => {
+    let layout = layoutView;
+    if (event.target.tagName && event.target.tagName.toLowerCase() === 'button') {
+      layout = event.target.getAttribute('data-layout');
+    }
 
-  const switchViews = layout => {
     const newUrl = new URL(window.location.href);
     newUrl.searchParams.set('layout', layout);
-    window.location.href = newUrl.href;
+    window.history.pushState({}, '', newUrl);
+    listingPageContent.classList.remove('wp-block-query--grid', 'wp-block-query--list');
+    listingPageContent.classList.add(`wp-block-query--${layout}`);
+
+    localStorage.setItem('layout', layout);
+    clearLocalStorage();
+
+    if (layout === 'list') {
+      toggleButton.title = __('Grid View', 'planet4-master-theme');
+      toggleButton.classList.remove('layout-toggle-grid', 'layout-toggle-list');
+      toggleButton.classList.add('layout-toggle-grid');
+      toggleButton.setAttribute('data-layout', 'grid');
+      toggleButton.setAttribute('aria-label', __('Switch to grid view', 'planet4-master-theme'));
+    }
+
+    if (layout === 'grid') {
+      toggleButton.title = __('List View', 'planet4-master-theme');
+      toggleButton.classList.remove('layout-toggle-grid', 'layout-toggle-list');
```

Comment:
Same doubt of [here ](https://github.com/greenpeace/planet4-master-theme/pull/2563/files#r2018684385) but poinitng the `layout-toggle-list`

Reply:
Same comment as above, its just a safety measure.